src/credentials/providers/credentials.ts

```diff
@@ -0,0 +1,120 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as telemetry from '../../shared/telemetry/telemetry.gen'
+
+const CREDENTIALS_PROVIDER_ID_SEPARATOR = ':'
+
+/**
+ * "Fully-qualified" credentials structure (source + name).
+ */
+export interface CredentialsId {
+    /** Credentials source id, e.g. "sharedCredentials". */
+    readonly credentialSource: CredentialsProviderType
+    /** User-defined profile name, e.g. "default". */
+    readonly credentialTypeId: string
+}
+
+/**
+ * Gets the string form of the given `CredentialsProvider`.
+ *
+ * For use in e.g. the statusbar, menus, etc.  Includes:
+ * - credentials source kind
+ * - instance-identifying information (typically the "profile name")
+ *
+ * @param credentials  Value to be formatted.
+ */
+export function asString(credentials: CredentialsId): string {
+    return [credentials.credentialSource, credentials.credentialTypeId].join(
+        CREDENTIALS_PROVIDER_ID_SEPARATOR
+    )
+}
+
+export function fromString(credentials: string): CredentialsId {
+    const separatorPos = credentials.indexOf(CREDENTIALS_PROVIDER_ID_SEPARATOR)
+
+    if (separatorPos === -1) {
+        throw new Error(`Unexpected credentialsId format: ${credentials}`)
+    }
+
+    const credSource = credentials.substring(0, separatorPos)
+    if (!credentialsProviderType.includes(credSource as any)) {
+        throw new Error(`unexpected credential source: ${credSource}`)
+    }
+
+    return {
+        credentialSource: credSource as CredentialsProviderType,
+        credentialTypeId: credentials.substring(separatorPos + 1),
+    }
+}
+
+export function isEqual(idA: CredentialsId, idB: CredentialsId): boolean {
+    return idA.credentialSource === idB.credentialSource && idA.credentialTypeId === idB.credentialTypeId
+}
+
+
+/**
+ * Credentials source type, broadly describes the kind of thing that supplied the credentials.
+ * 
+ * - "profile" is the most common, this is understood as a profile defined in
+ *   `~/.aws/credentials` or something like it, which may be authenticated via
+ *   static values, SSO, MFA, etc.
+ * - "ec2" or "ecs" means "instance credentials" given by the EC2/ECS "metadata service"
+ *   magic endpoint.
+ *   https://docs.aws.amazon.com/sdkref/latest/guide/setting-global-credential_source.html
+ * - "env" means we read the credentials from `AWS_XX` environment variables.
+ *   https://docs.aws.amazon.com/sdkref/latest/guide/environment-variables.html
+ *
+ * Compare the similar concept `telemetry.CredentialSourceId`.
+ */
+export type CredentialsProviderType = typeof credentialsProviderType[number];
+export const credentialsProviderType = ['profile', 'ec2', 'ecs', 'env'] as const;
```

Comment:
This is a new type. Unfortunately, although it is analogous to `telemetry.CredentialSourceId`, we don't want to use the telemetry type for [cosmetic and legacy purposes](https://github.com/aws/aws-toolkit-vscode/pull/1725#discussion_r626204148). 

Aside: we should have defined a common type in https://github.com/aws/aws-toolkit-common , in a common library, and used _that_ in telemetry. Not the other way around.